### ISSUE TITLE
Windows/MSVC/C++17 update

### ISSUE THREAD

**Issue by greenozon** [NotRequiem/VMAware#35]
All of a sudden was iterating over different C++ standards and hit this case when selected MSVC C++17:

```
The contents of <bit> are available only with C++20 or later.
```

inspecting the <bit> header in my system I've found the following:

![image](https://github.com/kernelwernel/VMAware/assets/765343/1598acdc-897e-43aa-99b6-123fa8937c1b)

as well as some official [docs ](https://en.cppreference.com/w/cpp/header/bit)stating that <bin> header is indeed part of C++20 standard

Here is a small PR https://github.com/kernelwernel/VMAware/pull/34

also if you don't mind I've added some printouts to end user showing the currently selected compiler standard as well as added support for C++23 standard which was already settled down:

__cplusplus  202302L

Do yo have some plans to start using features/headers from C++23?

**Comment by kernelwernel** [comment]
Ouch, that's a mistake on my part. Thanks for catching that! :)

- - -

> also if you don't mind I've added some printouts to end user showing the currently selected compiler standard as well as added support for C++23 standard which was already settled down

That's a pretty good idea, but I've modified it so that it works in a debug build instead of every build version. (note: the cmake debug build  is available with `cmake -DCMAKE_BUILD_TYPE=Debug` fyi)

- - -

> Do yo have some plans to start using features/headers from C++23?

Maybe, I'm not too familiar with C++23 if I'm being completely honest. I'll try to look around in the C++ standard's docs if I find something interesting to add though.

btw, the PR has been updated with [this](https://github.com/kernelwernel/VMAware/pull/34/commits/78deed6f5115fe9926f6d2d12dabd4dfe17de01e) commit.
